dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -472,32 +473,33 @@ public CloseableClient build() {
-                            TlsProtocols.get(),
-                            supportedCipherSuites(CipherSuites.allCipherSuites(), rawSocketFactory, name),
-                            new InstrumentedHostnameVerifier(
-                                    new DefaultHostnameVerifier(), name, conf.taggedMetricRegistry())) {
-                        @Override
-                        public Socket createSocket(final HttpContext context) throws IOException {
-                            return socksProxyAddress == null
-                                    ? super.createSocket(context)
-                                    : new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
-                        }
-                    };
+            Supplier<Socket> simpleSocketCreator = socksProxyAddress == null
+                    ? () -> new Socket(Proxy.NO_PROXY)
+                    : () -> new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
 
             PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager(
                     RegistryBuilder.<ConnectionSocketFactory>create()
                             .register(URIScheme.HTTP.id, new PlainConnectionSocketFactory() {
                                 @Override
-                                public Socket createSocket(final HttpContext context) throws IOException {
-                                    return socksProxyAddress == null
-                                            ? super.createSocket(context)
-                                            : new Socket(new Proxy(Proxy.Type.SOCKS, socksProxyAddress));
+                                public Socket createSocket(HttpContext _context) {
+                                    return simpleSocketCreator.get();
                                 }
                             })
-                            .register(URIScheme.HTTPS.id, sslSocketFactory)
+                            .register(
+                                    URIScheme.HTTPS.id,
+                                    new SSLConnectionSocketFactory(
+                                            MetricRegistries.instrument(
+                                                    conf.taggedMetricRegistry(), rawSocketFactory, name),
+                                            TlsProtocols.get(),
+                                            supportedCipherSuites(
+                                                    CipherSuites.allCipherSuites(), rawSocketFactory, name),
+                                            new InstrumentedHostnameVerifier(
+                                                    new DefaultHostnameVerifier(), name, conf.taggedMetricRegistry())) {
+                                        @Override
+                                        public Socket createSocket(HttpContext _context) {
+                                            return simpleSocketCreator.get();
+                                        }
```

Comment:
Updated to use the supplier, but also moved the implementation inline here matching `PlainConnectionSocketFactory` -- it was odd that one was defined inline, and the other was not.